[CI] Zen1IT testMixedClusterFormation failing

Likely relates the backport of #76771

**Build scan:**
https://gradle-enterprise.elastic.co/s/kw3zm7ec5erls/tests/:server:internalClusterTest/org.elasticsearch.cluster.coordination.Zen1IT/testMixedClusterFormation

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.coordination.Zen1IT.testMixedClusterFormation" -Dtests.seed=59A2DAAB29E9A6F6 -Dtests.locale=uk-UA -Dtests.timezone=America/Monterrey -Druntime.java=8`

**Applicable branches:**
7.x

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.Zen1IT&tests.test=testMixedClusterFormation

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=3751, name=elasticsearch[node_t2][masterService#updateTask][T#1], state=RUNNABLE, group=TGRP-Zen1IT]


  Caused by: java.lang.AssertionError: (No message provided)

    at __randomizedtesting.SeedInfo.seed([59A2DAAB29E9A6F6]:0)
    at org.elasticsearch.cluster.ClusterStatePublicationEvent.getPublicationContextConstructionElapsedMillis(ClusterStatePublicationEvent.java:95)
    at org.elasticsearch.cluster.service.MasterService$ClusterStateUpdateStatsTracker.onPublicationSuccess(MasterService.java:953)
    at org.elasticsearch.cluster.service.MasterService.onPublicationSuccess(MasterService.java:316)
    at org.elasticsearch.cluster.service.MasterService.publish(MasterService.java:293)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:267)
    at org.elasticsearch.cluster.service.MasterService.access$100(MasterService.java:63)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:161)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:678)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:259)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:222)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/Zen1IT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/Zen1IT.java
@@ -61,15 +61,14 @@ import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/76847")
 public class Zen1IT extends ESIntegTestCase {
 
-    private static Settings ZEN1_SETTINGS = Coordinator.addZen1Attribute(true, Settings.builder()
+    private static final Settings ZEN1_SETTINGS = Coordinator.addZen1Attribute(true, Settings.builder()
         .put(DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey(), DiscoveryModule.ZEN_DISCOVERY_TYPE)
         .put(IndicesService.WRITE_DANGLING_INDICES_INFO_SETTING.getKey(), false)
         ).build();
 
-    private static Settings ZEN2_SETTINGS = Settings.builder()
+    private static final Settings ZEN2_SETTINGS = Settings.builder()
         .put(DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey(), DiscoveryModule.ZEN2_DISCOVERY_TYPE)
         .build();
 

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/Zen1IT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/Zen1IT.java
@@ -7,7 +7,6 @@
  */
 package org.elasticsearch.cluster.coordination;
 
-import org.apache.lucene.util.LuceneTestCase;
 import org.elasticsearch.action.admin.cluster.configuration.AddVotingConfigExclusionsAction;
 import org.elasticsearch.action.admin.cluster.configuration.AddVotingConfigExclusionsRequest;
 import org.elasticsearch.action.admin.cluster.configuration.ClearVotingConfigExclusionsAction;

--- a/server/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java
+++ b/server/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java
@@ -317,6 +317,12 @@ public class ZenDiscovery extends AbstractLifecycleComponent implements Discover
         assert newState.getNodes().isLocalNodeElectedMaster()
             : "Shouldn't publish state when not master " + clusterStatePublicationEvent.getSummary();
 
+        // don't record timing stats in legacy discovery
+        clusterStatePublicationEvent.setPublicationContextConstructionElapsedMillis(0L);
+        clusterStatePublicationEvent.setPublicationCommitElapsedMillis(0L);
+        clusterStatePublicationEvent.setMasterApplyElapsedMillis(0L);
+        clusterStatePublicationEvent.setPublicationCompletionElapsedMillis(0L);
+
         try {
 
             // state got changed locally (maybe because another master published to us)

